src/paradigma/heart_rate/heart_rate_estimation.py

```diff
@@ -0,0 +1,185 @@
+import numpy as np
+from typing import Tuple
+
+from paradigma.config import HeartRateExtractionConfig
+from paradigma.heart_rate.tfd import nonsep_gdtfd
+
+def assign_sqa_label(ppg_prob, config: HeartRateExtractionConfig, acc_label=None) -> np.ndarray:
+    """
+    Assigns a signal quality label to every individual data point.
+
+    Parameters:
+    - ppg_prob: numpy array of probabilities for PPG
+    - acc_label: numpy array of labels for accelerometer (optional)
+    - fs: Sampling frequency
+
+    Returns:
+    - sqa_label: numpy array of signal quality assessment labels
+    """
+    # Default _label to ones if not provided
+    if acc_label is None:
+        acc_label = np.ones(len(ppg_prob))
+
+
+    # Number of samples in an epoch
+    fs = config.sampling_frequency
+    samples_per_epoch = config.window_length_s * fs
+
+    # Calculate number of samples to shift for each epoch
+    samples_shift = config.window_step_size_s * fs
+    n_samples = int((len(ppg_prob) + config.window_overlap_s) * fs)
+    data_prob = np.zeros(n_samples)
+    data_label_imu = np.zeros(n_samples, dtype=np.int8)
+
+    for i in range(n_samples):
+        # Start and end indices for current epoch
+        start_idx = int(np.floor((i - (samples_per_epoch - samples_shift)) / fs))
+        end_idx = int(np.floor(i / fs))
```

Comment:
I believe `int` already does `np.floor` inherently, but you could check.

Reply:
Not for negative numbers, and that's what I need, so I will update it to integer division because that's probably the most efficient

Reply:
Could `i - (samples_per_epoch - samples_shift)` be negative? Smart thinking though about the `np.floor` for negative values.

Reply:
Yes it is for the edge cases at the start where i = 1:150 (because samples_per_epoch is 180 and samples_shift is 30)